### ISSUE TITLE
Simulate shield destruction

### ISSUE THREAD

**Issue by tonyndory** [xosofox/IPAS#33]
When attacking a portal in game, there is a chance of destroying the shields with each XMP used.  Is it possible to incorporate this into the simulator?  Testing would need to be done to discover the percentage chance bursters have for destroying shields, but this would be incredibly useful.

**Comment by Hackswell** [comment]
Any progress on data collection for XMPs and US?

**Comment by tonyndory** [comment]
No, I have basically been using my subjective experience as my guide.

**Comment by Hackswell** [comment]
Xosofox, how does one get accurate numbers in order to help you with this?

**Comment by Xosofox** [comment]
The only "reliable" way is to intercept the traffic between client and
server and store and then evaluate the results - which of course nobody
would do as this most likely violates the ToS

2015-04-01 19:00 GMT+02:00 Sir Woody Hackswell <notifications@github.com>:

**Comment by lablua** [comment]
It's really not that hard to do - configure your Android device to talk to a common SSL MITM proxy like Charles, and you can spy on the client/server communication. It'll somewhat increase your server lag, but otherwise have no negative effects.

I'm not sure it would violate the ToS - at least reading it, I cannot see any paragraph it would clearly violate, but IANAL. It doesn't seem like the needed information is within the definition of "Content", but if it is then you would run afoul of paragraph 2(d). 

Paragraph 2(c) is hard to say either way, as it is phrased so broadly one could argue Project Loon is the only compliant way to use Ingress, since all the cell networks involve access through technology not provided by Google. I don't see how using a proxy server you own is any worse than using your own wifi hotspot.

**Comment by Xosofox** [comment]
> It's really not that hard to do
I never said it would be. Collecting the data is not the problem if you have time and determination.

But reverse-engineering the algorithm that calculates the shield-pop-probability is another story

**Comment by lablua** [comment]
Hard to know how tough that is until you have the data, and run it through some statistical analysis. Unless someone has tried and failed, we just don't know. If they have, I'd like to see the anonymized data published so more people can try to analyze it.

It *could* be a simple probability calculated in an obvious manner from the "stickiness" property of a shield, or it could be something far more complex like shields accumulating damage over the lifetime the mod is on the portal. If it's the former, a few hundred samples per weapon and shield combination could be enough to figure it out. If it's the latter, it might require controlled experiments with viruses and multiple agents firing to get the necessary data.
